test/Microsoft.AspNetCore.SignalR.Client.Tests/HttpConnectionTests.ConnectionLifecycle.cs

```diff
@@ -168,11 +178,38 @@ Task OnTransportStart()
                         {
                             var ex = await Assert.ThrowsAsync<InvalidOperationException>(() => connection.StartAsync(TransferFormat.Text));
                             Assert.Equal("Unable to connect to the server with any of the available transports.", ex.Message);
-                            Assert.Equal(3, startCounter);
+
+                            // If websockets aren't supported then we expect one less attmept to start.
+                            if (!IsWebSocketsSupported())
+                            {
+                                Assert.Equal(availableTransports - 1, startCounter);
+                                return;
+                            }
+                            Assert.Equal(availableTransports, startCounter);
                         });
                 }
             }
 
+            private static bool IsWebSocketsSupported()
```

Comment:
Can we use the code from the condition attribute @JamesNK added? Put it in a helper method or something?